tools/build_file_generator/BuildFileGenerator.java

```diff
@@ -71,52 +76,78 @@ private ThirdPartyDeps getThirdPartyDeps() {
             THIRD_PARTY_ZIP_PATH, PROTOTXT_FILENAME_INSIDE_ZIP, ThirdPartyDeps.newBuilder());
   }
 
-  BuildFile generateBuildFile(String packagePath) throws IOException {
-    BuildFile.Builder result = BuildFile.newBuilder();
+  private List<String> getPathsToGenerateBuildFiles() {
+    List<String> result = new ArrayList<>();
+    try {
+      for (String item :
+          fileUtils.listContentsRecursively(fileUtils.getCurrentWorkingDirectory())) {
+        if (fileUtils.folderExists(item)) {
+          List<String> folderContent = fileUtils.listContents(item);
+          for (String fileName : folderContent) {
+            if (fileName.endsWith(".java") || fileName.endsWith(".proto")) {
+              result.add(item);
+            }
+          }
+        }
+      }
+    } catch (IOException e) {
+      e.printStackTrace();
+    }
+    return result;
+  }
+
+  Map<String, BuildFile> generateBuildFiles() throws IOException {
```

Comment:
I think it was actually nice to have a method for `generateBuildFile()` that gets called by `generateBuildFiles()`, especially since it's already quite a large method, so that would sort of separate the "single element" logic from the 'apply it to a list" logic.

Reply:
Done.